### DISCUSSION TITLE
Mouse scrolling works when it wants to

### DISCUSSION THREAD

**Discussion by JaggedOne** [induktio/thinker#67] [Q&A]
Man, I love mouse scrolling in the F4 window; once I get more than a full screen of installations, it is just heaven. The problem is that it only works sometimes, I mean, I refrained from saying that it only works when it wants to, but sheesh. I have no idea what the criteria is that governs this. Today I continued my current game, and it wasn't working although it did last night. To test it, I exited and then immediately re-ran the game, and it worked. This is driving me nutters !!!

For the record, I am running Windows 7 Home Premium, 64-bit, with Service Pack 1 and a Logitech mouse.

**Comment by induktio** [comment]
In those instances where mouse scrolling doesn't work, does anything happen when you press ALT+T? These things might be related since there's some unspecified startup issue that happens only randomly where the game doesn't register the keyboard shortcut handlers for the mod, and therefore ALT+T doesn't work. If anyone has insight on how to patch the issue that would be useful, but so far the only workaround is to restart the game.

**Reply by JaggedOne** [reply]
Interesting !!! It took several stops/starts over the course of today before I got a game where the F4 box wouldn't scroll. I then tried ALT-T, and found that it did NOT bring up the Thinker menu like it normally did, it just performed the 'T' action that strips off a layer of structure (not sure what that's officially called). I exited, re-ran the game, and all was fine.

On a very interesting note - gotta be related: I periodically have to exit the game to check my browser, email, etc, because I cannot ALT-TAB back to Windows from within the game (I have been meaning to post a question about that - the GOG instructions for enabling ALT-TAB from within one of their games does not work for me - I have posted on their support forum, no response so far). Anyway, at one point this evening, I exited SMAC to check my email and discovered that ALT-TAB wouldn't work in Windows !!! After trying some things, I power-cycled (not Restarted) my system, and when it came back up, things were back to normal (I good ole power cycle fixes an astoundingly high percentage of problems, not just in computers, either).

Anyway, not sure what to make of all this info, just passing it on....

**Reply by induktio** [reply]
Looks like the issue was found and there's now a new build available (Apr 7) that includes some changes in the startup code to fix it. The issue was a race condition in DLL initialization after all. There's also new implementation for the mouse wheel scrolling issue, so now the code should always properly detect if the world map is active or not.
